rubygem-smart_proxy_dynflow_core/rubygem-smart_proxy_dynflow_core.spec

```diff
@@ -0,0 +1,142 @@
+Requires: %{?scl_prefix_ror}rubygem(sqlite3)
+Requires: %{?scl_prefix_ruby}ruby(release)
+Requires: %{?scl_prefix_ruby}rubygems
+%if 0%{?rhel} == 6
+Requires(post): chkconfig
+Requires(preun): chkconfig
+Requires(preun): initscripts
+%else
+Requires(post): systemd-sysv
+Requires(post): systemd-units
+Requires(preun): systemd-units
+BuildRequires: systemd
+%endif
+
+BuildRequires: %{?scl_prefix_ruby}ruby(release)
+BuildRequires: %{?scl_prefix_ruby}rubygems
+BuildRequires: %{?scl_prefix_ruby}rubygems-devel
+
+BuildArch: noarch
+Provides: %{?scl_prefix}rubygem(%{gem_name}) = %{version}
+
+%description
+Use the Dynflow inside Foreman smart proxy
+
+%package doc
+BuildArch:  noarch
+Requires:   %{?scl_prefix}%{pkg_name} = %{version}-%{release}
+Summary:    Documentation for rubygem-%{gem_name}
+
+%description doc
+This package contains documentation for rubygem-%{gem_name}.
+
+%prep
+%setup -n %{pkg_name}-%{version} -q -c -T
+mkdir -p .%{gem_dir}
+%{?scl:scl enable %{scl} "}
+gem install --local --install-dir .%{gem_dir} \
+            --force %{SOURCE0} --no-rdoc --no-ri
+%{?scl:"}
+
+%build
+sed -ri '1sX(/usr/bin/ruby|/usr/bin/env ruby)X/usr/bin/%{?scl:%{scl_prefix}}rubyX' %{_builddir}/%{pkg_name}-%{version}/%{gem_instdir}/bin/%{service_name}
+
+# switches to bundler_ext instead of bundler and uses bundler.d from /etc/foreman-proxy/dynflow/bundler.d
+mv %{_builddir}/%{pkg_name}-%{version}/%{gem_instdir}/Gemfile %{_builddir}/%{pkg_name}-%{version}/%{gem_instdir}/Gemfile.in
+sed -ri 'sX\#\{File\.dirname\(__FILE__\)\}/bundler\.d/\*\.rbX%{_sysconfdir}/foreman-proxy/dynflow/bundler.d/*.rbX' %{_builddir}/%{pkg_name}-%{version}/%{gem_instdir}/Gemfile.in
+
+%install
+mkdir -p %{buildroot}%{_sysconfdir}/foreman-proxy/dynflow/bundler.d
+mkdir -p %{buildroot}%{gem_dir}
+cp -a .%{gem_dir}/* \
+        %{buildroot}%{gem_dir}/
+
+#copy init scripts and sysconfigs
+%if 0%{?rhel} == 6
+install -Dp -m0755 %{buildroot}%{gem_instdir}/deploy/%{service_name}.init %{buildroot}%{_root_initddir}/%{service_name}
+%else
+install -Dp -m0644 %{buildroot}%{gem_instdir}/deploy/%{service_name}.service %{buildroot}%{_unitdir}/%{service_name}.service
+%endif
+mkdir -p %{buildroot}%{_scl_root}%{bin_dir}
+ln -sv %{gem_instdir}/bin/%{service_name} %{buildroot}%{_scl_root}%{bin_dir}/%{service_name}
+
+%post
+%if 0%{?rhel} == 6
+  /sbin/chkconfig --add %{service_name}
+  exit 0
+%else
+  %systemd_post %{service_name}.service
+%endif
+
+%preun
+%if 0%{?rhel} == 6
+  if [ $1 -eq 0 ] ; then
+    /sbin/service %{service_name} stop >/dev/null 2>&1
+    /sbin/chkconfig --del %{service_name}
+  fi
+%else
+  %systemd_preun %{service_name}.service
+%endif
+
+%postun
+%if 0%{?rhel} != 6
+  %systemd_postun_with_restart %{service_name}.service
+%endif
+
+%files
+%dir %{gem_instdir}
+%dir %{_sysconfdir}/foreman-proxy/dynflow/bundler.d
+%exclude %{gem_instdir}/.*
+%{gem_libdir}
+%{gem_instdir}/bin
+%{gem_instdir}/config
+%{gem_instdir}/Gemfile.in
+%{gem_instdir}/smart_proxy_dynflow_core.gemspec
```

Comment:
Oh sorry, I know why it's a gem.  If you're using a newer ruby, and smart_proxy_dynflow and smart_proxy_dynflow_core are running in the same ruby environment (say on el7 where the ruby is new enough), then you don't actually need to use the separate service.

We're not *currently* using that feature, but could in the future.